pkg/client/src/app/common/KeycloakProvider.tsx

```diff
@@ -18,63 +19,107 @@ export const KeycloakProvider: React.FunctionComponent<
   // TODO: Implement a short circuit for overriding auth requirements
   //   const overrideAuth = process.env["AUTH_REQUIRED"];
 
-  return (
-    <>
-      <ReactKeycloakProvider
-        authClient={keycloak}
-        initOptions={{ onLoad: "login-required" }}
-        LoadingComponent={
-          <Flex
-            spaceItems={{ default: "spaceItemsSm" }}
-            alignItems={{ default: "alignItemsCenter" }}
-            flexWrap={{ default: "nowrap" }}
-            style={{
-              width: "100%",
-              height: "100%",
-            }}
-          >
-            <FlexItem
+  const setCookie = (cName: string, cValue: string, expDays: number) => {
+    let date = new Date();
+    date.setTime(date.getTime() + expDays * 24 * 60 * 60 * 1000);
+    const expires = "expires=" + date.toUTCString();
+    document.cookie = `${cName} = ${cValue}; ${expires}; path=/`;
+  };
+
+  const getCookie = (token: string) => {
+    let cookieArr = document.cookie.split(";");
+    for (let i = 0; i < cookieArr.length; i++) {
+      let cookiePair = cookieArr[i].split("=");
+      if (token == cookiePair[0].trim()) {
+        return decodeURIComponent(cookiePair[1]);
+      }
+    }
+    return null;
+  };
+
+  const checkCookie = () => {
+    let token = getCookie("proxyToken");
+    if (token !== "" && token !== null) {
+    } else {
+      token = keycloak?.token || "";
+
+      if (token != "" && token != null) {
+        setCookie("proxyToken", token, 365);
+      }
+    }
+  };
+  const deleteCookie = (name: string) => {
+    document.cookie = `${name} =; Path=/; Expires=Thu, 01 Jan 1970 00:00:01 GMT;`;
+  };
+  useEffect(() => {
+    deleteCookie("proxyToken");
+    return () => {
+      checkCookie();
+    };
+  }, [keycloak.token]);
+  const isAuthRequired = ENV.AUTH_REQUIRED !== "false";
+  if (isAuthRequired) {
+    return (
+      <>
+        <ReactKeycloakProvider
+          authClient={keycloak}
+          initOptions={{ onLoad: "login-required" }}
+          LoadingComponent={
+            <Flex
+              spaceItems={{ default: "spaceItemsSm" }}
+              alignItems={{ default: "alignItemsCenter" }}
+              flexWrap={{ default: "nowrap" }}
               style={{
-                margin: "auto auto",
-                textAlign: "center",
+                width: "100%",
+                height: "100%",
```

Comment:
Let's use PF css approach for consistency.

Reply:
This PR doesn't touch this code. Just bumps the block down due to adding a wrapper around the rendered jsx.